scripts/generate_ironside_ipc_request.py

```diff
@@ -0,0 +1,320 @@
+
+    Returns:
+        (ipc_request_data, memory_blob_data)
+    """
+
+    def parse_hex_string(hex_str: str) -> bytes:
+        return binascii.unhexlify(hex_str.removeprefix("0x"))
+
+    psa_key_attrs_data = parse_hex_string(psa_key_attrs_hex)
+    key_data = parse_hex_string(key_value_hex)
+
+    return generate_psa_ipc_request(
+        function_id=TFM_CRYPTO_IMPORT_KEY_SID,
+        input_data_list=[psa_key_attrs_data, key_data],
+        output_specs=[(key_id_address, key_id_len)],
+        base_address=base_address,
+    )
+
+
+def generate_random_request(
+    output_address: int,
+    output_length: int,
+    base_address: int,
+) -> tuple[bytes, bytes]:
+    """
+    Generate IPC request data for the psa_generate_random function.
+
+    Returns:
+        (ipc_request_data, memory_blob_data)
+    """
+    # psa_generate_random only needs the output buffer, no input data beyond iovec
+    return generate_psa_ipc_request(
+        function_id=TFM_CRYPTO_GENERATE_RANDOM_SID,
+        input_data_list=[],  # No additional input data beyond iovec
+        output_specs=[(output_address, output_length)],
+        base_address=base_address,
+    )
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Generate IPC request binary blobs for PSA operations",
+        allow_abbrev=False,
+    )
+
+    parser.add_argument(
+        "--base-address",
+        help="Base address for memory blob",
+        type=lambda x: int(x, 0),
+        required=True,
+    )
+
+    parser.add_argument(
+        "--output",
+        help="Output JSON filename (if not provided, outputs Intel HEX to stdout)",
+        type=argparse.FileType("wb"),
+        required=False,
+    )
+
+    subparsers = parser.add_subparsers(
+        dest="operation",
+        help="IPC operation to perform",
+        required=True,
+    )
+
+    import_parser = subparsers.add_parser(
+        "psa_import_key",
+        help="Invoke psa_import_key through the PSA service",
+    )
+
+    import_parser.add_argument(
+        "--psa-key-attrs",
+        help="PSA key attributes as hex string. This can be found by running the "
+        "generate_psa_key_attributes.py script",
+        type=str,
+        required=True,
+    )
+
+    import_parser.add_argument(
+        "--key-value",
+        help="Key value as hex string",
+        type=str,
+        required=True,
+    )
+
+    import_parser.add_argument(
+        "--key-id-address",
+        help="Base address for key ID output buffer",
+        type=lambda x: int(x, 0),
+        required=True,
+    )
+
+    import_parser.add_argument(
+        "--key-id-len",
+        help="Length of key ID output buffer",
+        type=lambda x: int(x, 0),
+        required=True,
+    )
+
+    random_parser = subparsers.add_parser(
+        "psa_generate_random",
+        help="Invoke psa_generate_random through the PSA service",
+    )
+
+    random_parser.add_argument(
+        "--output-address",
```

Comment:
Clarify that this cannot collide with the `base_address` and come with examples of use.